### PR TITLE
Build OTel Agent on CI

### PR DESCRIPTION
**Pull request by alopezz**
### What does this PR do?

Adds a CI job for building of the OTel Agent binary.

### Motivation

[OASIS-7](https://datadoghq.atlassian.net/browse/OASIS-7)

### Additional Notes

The build result is currently uploaded as a GitLab artifact, but we may iterate on this depending on still evolving needs.

### Describe how to test/QA your changes

[OASIS-7]: https://datadoghq.atlassian.net/browse/OASIS-7?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM as long as the CI passes !

**Review by spencergilbert — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

File: .gitlab/binary_build/otel_agent.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+otel_agent_binary_build:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
```
Comment (dineshg13) [inline .gitlab/binary_build/otel_agent.yml:4]: We would need to add `arm64` build. Internally staging cluster using ARM

**Review by dineshg13 — CHANGES_REQUESTED** [review]

File: .gitlab/binary_build/otel_agent.yml (hunk 1 above)
Comment (alopezz) [reply .gitlab/binary_build/otel_agent.yml:4]: Sounds good, I'll add that 👍

File: .gitlab/binary_build/otel_agent.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.build_otel_agent_binary_common:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  stage: binary_build
+  rules:
+    - !reference [.except_mergequeue]
+    - when: on_success
+  script:
+    - !reference [.retrieve_linux_go_deps]
+    - inv -e otel-agent.build
+  needs: ["go_deps"]
```
Comment (dineshg13) [inline .gitlab/binary_build/otel_agent.yml:11]: Should we add lint and go mod tidy checks ?

Comment (alopezz) [reply .gitlab/binary_build/otel_agent.yml:11]: We certainly can, I don't have much of an opinion about that. I've looked at other binary build jobs and they're a bit inconsistent in this regard. Note that the only effect of adding them here is that this job will wait until those other jobs finish successfully, which is not strictly necessary (whereas `go_deps` are definitely necessary I think).

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**